orchestrator/careplanservice/handle_get.go

```diff
@@ -0,0 +1,191 @@
+	"net/http"
+	"strings"
+)
+
+func (s *Service) handleGet(httpResponse http.ResponseWriter, request *http.Request) error {
+	queryParams := request.URL.Query()
+	urlParts := strings.Split(strings.TrimPrefix(request.URL.Path, basePath+"/"), "/")
+	switch urlParts[0] {
+	case "CarePlan":
+		// fetch CarePlan + CareTeam, validate requester is participant of CareTeam
+		// if there are query params present, we need to handle the request a bit differently
+		if len(queryParams) != 0 {
+			if queryParams.Get("_id") == "" {
+				return coolfhir.NewErrorWithCode("Query parameter _id is not present, can't find CarePlan", http.StatusBadRequest)
+			}
+
+			var bundle fhir.Bundle
+			err := s.fhirClient.Read("CarePlan", &bundle, fhirclient.QueryParam("_id", queryParams.Get("_id")), fhirclient.QueryParam("_include", "CarePlan:care-team"))
+			if err != nil {
+				return err
+			}
+			var careTeams []fhir.CareTeam
+			err = coolfhir.ResourcesInBundle(&bundle, coolfhir.EntryIsOfType("CareTeam"), &careTeams)
+			if err != nil {
+				return err
+			}
+			if len(careTeams) == 0 {
+				return coolfhir.NewErrorWithCode("CareTeam not found in bundle", http.StatusNotFound)
+			}
+
+			err = validatePrincipalInCareTeams(request.Context(), careTeams)
+			if err != nil {
+				return err
+			}
+
+			// Valid requester, now perform the bundle request again with all the params
+			params := []fhirclient.Option{}
+			for k, v := range queryParams {
+				params = append(params, fhirclient.QueryParam(k, v[0]))
+			}
+			err = s.fhirClient.Read("CarePlan", &bundle, params...)
+			if err != nil {
+				return err
+			}
+
+			b, err := json.Marshal(bundle)
+			_, err = httpResponse.Write(b)
+			if err != nil {
+				return err
+			}
+			return nil
+		}
+		if len(urlParts) < 2 {
+			return &coolfhir.ErrorWithCode{
+				Message:    "URL missing CarePlan ID",
+				StatusCode: http.StatusForbidden,
+			}
+		}
+		carePlan, careTeams, err := s.getCarePlanAndCareTeams("CarePlan/" + urlParts[1])
+		if err != nil {
+			return err
+		}
+
+		err = validatePrincipalInCareTeams(request.Context(), careTeams)
+		if err != nil {
+			return err
+		}
+
+		b, err := json.Marshal(carePlan)
+		_, err = httpResponse.Write(b)
+		if err != nil {
+			return err
+		}
+		// TODO: Do we support additional info in the URL i.e. extra params
+	case "CareTeam":
+		// fetch CareTeam, validate requester is participant
+		if len(urlParts) < 2 {
+			return &coolfhir.ErrorWithCode{
+				Message:    "URL missing CareTeam ID",
+				StatusCode: http.StatusForbidden,
+			}
+		}
+		var careTeam fhir.CareTeam
+		err := s.fhirClient.Read("CareTeam/"+urlParts[1], &careTeam)
+		if err != nil {
+			return err
+		}
+		err = validatePrincipalInCareTeams(request.Context(), []fhir.CareTeam{careTeam})
+		if err != nil {
+			return err
+		}
+
+		b, err := json.Marshal(careTeam)
+		_, err = httpResponse.Write(b)
+		if err != nil {
+			return err
+		}
+		// TODO: Do we support additional info in the URL i.e. extra params
```

Comment:
Caller won't get the headers returned by the FHIR API. You can capture these with `fhirclient.ResponseHeaders(headers)` option for `Read()`, then set them on the HTTP response;
```
		for key, value := range headers.Header {
			httpResponse.Header()[key] = value
		}
```

Reply:
(I took this from `lib/coolfhir/bundle.go:195`)

Reply:
(also applies to the other handlers in this file)

Reply:
Looks good, thanks!